themes/default/theme/stencil/src/components/ai/ai.tsx

```diff
@@ -316,6 +382,14 @@ export class PulumiAI {
                         <span slot="content">${ this.copyButtonTooltip }</span>
                     </pulumi-tooltip>
                 </li>
+                ${ shareable ? `<li class="ml-2">
+                    <pulumi-tooltip>
```

Comment:
So the way this shakes out is that the UI makes it look like what you're sharing is a code block, but what you're really sharing is effectively the whole page/conversation. In that sense, it seems like the share button might belong more at the "top level" somehow  -- although admittedly there doesn't seem to be a great place in the current layout that quite makes room for that. (At first I thought it might go better at the prompt level -- but that doesn't quite work either, as conversations can have multiple prompts.)

I guess it's worth asking what one would expect to be sharing here. If you'd expect to share the whole conversation, then we might want to come up with more of a top-level position for the button, if only to make it clearer to the user they'll be sharing the whole page. (For this, we might even consider updating the URL dynamically with the conversation ID and just letting users copy and share that.) For my part, I sort of imagined this working by scoping the share to the "prompt + result combo" level, in part because conversations can be iterative (and I'd probably be most interested in sharing the end result) ... but that's just me; others might want to use this differently. What do you think?